### PR TITLE
INT B-20591 prime can update destination address for NTSR shipment

### PR DESCRIPTION
**Pull request by danieljordan-caci**
## Summary

The gov is wanting NTSR shipments to go through the same process as HHG shipments when the Prime requests a change to the delivery/destination address with the end goal being the TOO needs to approve the change if the business rules change.

This PR does the following:
- some random white space deletions cause it was driving me nuts
- added on to existing query when looking for shipments that are requesting an addy change - for NTSR shipments we need the storage address since technically that's the "pickup" address
- adjusted logic checks to either look at pickup or storage address, depending on `shipmentType`
- updated UI to conditionally show `Review required` instead of the address for NTSR shipments that have a pending destination address change
- added a way for the office user to click the `See delivery location to proceed.` link and be scrolled to where they need their attention
- disabled destination fields when the NTSR shipment has a pending destination address request
- let the TOO open up the destination address request modal for NTSR shipments
- minor CSS changes
- updated/added relevant tests to both backend and UI

### How to test

1. Fire up yer `client` & `server`
2. Find any shipment as a TOO (my fav is `SITEXT`) and add on a NTSR shipment
3. Fill out all of the necessary information
4. Make sure you approve the shipment (this makes it available to the Prime)
5. Go into the Prime Simulator, find the shipment inside the move, and click the `Update Shipment Destination Address` button
6. Fill out the form with an address that's hecka far away from the current ZIP, save it
7. Go into the same move as the TOO and you should see the shipment have a `Review required` where the NTSR's destination address should be
8. Go through the flow and either approve/reject the shipment (approving changes the address)
9. Now go through steps again and change the ZIP to be somewhere or the same ZIP - you shouldn't need approval from the TOO now since the business rules don't change

## Screenshots
Screenshot 2024-07-29 at 11 08 58 AM

https://github.com/user-attachments/assets/42470d73-6ff0-4933-8bc9-0880e1813bf3

### CODE REVIEW

File: pkg/services/shipment_address_update/shipment_address_update_requester.go [hunk 1]
```diff
@@ -245,8 +245,8 @@ func (f *shipmentAddressUpdateRequester) RequestShipmentDeliveryAddressUpdate(ap
 	if shipment.MoveTaskOrder.AvailableToPrimeAt == nil {
 		return nil, apperror.NewUnprocessableEntityError("destination address update requests can only be created for moves that are available to the Prime")
 	}
-	if shipment.ShipmentType != models.MTOShipmentTypeHHG {
-		return nil, apperror.NewUnprocessableEntityError("destination address update requests can only be created for HHG shipments")
+	if shipment.ShipmentType != models.MTOShipmentTypeHHG && shipment.ShipmentType != models.MTOShipmentTypeHHGOutOfNTSDom {
+		return nil, apperror.NewUnprocessableEntityError("destination address update requests can only be created for HHG and NTSr shipments")
```
Comment (TevinAdams) [inline pkg/services/shipment_address_update/shipment_address_update_requester.go:249]: Is the r in NTSr meant to be lower case or did you mean NTSR?

**Comment by TevinAdams** [comment]
@danieljordan-caci I created an NTS shipment with zipcode 75126 TX, Approved as TOO, and then updated the destination address as the prime to 29229, SC. Went back to see the destination address status and noticed that destination address isn't showing up on the shipment like HHG shipments. Am I doing something incorrectly?
image

**Comment by danieljordan-caci** [comment]
> @danieljordan-caci I created an NTS shipment with zipcode 75126 TX, Approved as TOO, and then updated the destination address as the prime to 29229, SC. Went back to see the destination address status and noticed that destination address isn't showing up on the shipment like HHG shipments. Am I doing something incorrectly? image

Delivery address == destination address

**Comment by TevinAdams** [comment]
> > @danieljordan-caci I created an NTS shipment with zipcode 75126 TX, Approved as TOO, and then updated the destination address as the prime to 29229, SC. Went back to see the destination address status and noticed that destination address isn't showing up on the shipment like HHG shipments. Am I doing something incorrectly? image
> 
> Delivery address == destination address

Disregard i updated the wrong shipment.

File: pkg/services/shipment_address_update/shipment_address_update_requester.go (hunk 1 above)
Comment (danieljordan-caci) [reply pkg/services/shipment_address_update/shipment_address_update_requester.go:249]: It's sometimes referred to like this in the code and sometimes NTSR. Don't mind changing it if you hate it haha

Comment (TevinAdams) [reply pkg/services/shipment_address_update/shipment_address_update_requester.go:249]: Nope just making sure it was intended. Everything works and reviewed. Looks good.

File: swagger-def/prime.yaml [hunk 2]
```diff
@@ -550,7 +550,13 @@ paths:
         ### Functionality
         This endpoint is used so the Prime can request an **update** for the destination address on an MTO Shipment,
         after the destination address has already been approved.
-        If automatically approved or TOO approves, this will update the final destination address values for destination SIT service items to be the same as the changed destination address that was approved.
+
+        This should ONLY be used for the following shipment types:
+        - HHG
+        - NTSR
+
+        For HHG shipments, if automatically approved or TOO approves, this will update the final destination address values for destination SIT service items to be the same as the changed destination address that was approved.
```
Comment (cameroncaci) [inline swagger-def/prime.yaml:558]: This mentions "should" but appears to be an explicit requirement - please correct if wrong

**Review by cameroncaci — CHANGES_REQUESTED** [review]
Works like a charm, got a quick question on the API description though

File: swagger-def/prime.yaml (hunk 2 above)
Comment (cameroncaci) [reply swagger-def/prime.yaml:558]: Can we reword that to be "This endpoint and operation only supports the following shipment types:" 🫣

Comment (danieljordan-caci) [reply swagger-def/prime.yaml]: You want it, you go it! Updated. Resolving comment.

**Review by cameroncaci — APPROVED** [review]
Works locally and meets AC 🚀

**Pull request merged**